tests/Stub/Gutenberg/BlockParser.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WpOop\WordPress\Test\Stub\Gutenberg;
+
+use WpOop\WordPress\Gutenberg\BlockParserInterface;
+
+class BlockParser implements BlockParserInterface
```

Comment:
That's the thing, see: if you want to create stubs, you are going to add some kind of implementation. I understand that it's just for testing, but why not simply create a mock, like for all other interfaces in all other projects?

Reply:
It's simpler, less risk to add a mistake because of misuse of mocking lib etc. In this case there is no logic to mock.

Reply:
> like for all other interfaces in all other projects?

They only test instantiation. Here :arrow_down: 

>  they also test that we can actually e.g. call `parseBlocks` with a string and it can return an array of blocks.

Reply:
I know they test just instantiation. The point in question is: creating extra stubs, VS giving all control to the test itself. I prefer the latter.

Reply:
Creating such basic stubs seems much less complex than duplicating all that via mocking. And PHPUnit mocks seem to have less strict type checks, which is the main purpose of these tests, for example you call `$subject->getBaseName()` in the plugin test, while you will get an error if trying to return `null` instead of `string` in stubs.

Reply:
What is complicated about `$this->getMockBuilder(BlockParserInterface::class)->getMock()`? Why does it matter what happens when you call `$subject->getBaseName()`, if you are simply testing that the interface can be instantiated corretly, and not invocation of any methods?

Reply:
It is done to test that we do not have any weird mistakes in the function definitions, type-hints, etc., that they work as expected.

Reply:
The definitions, including hints, are already provided by the interface, and it therefore seems impossible to make a mistake there. However, the benefit is that the test is fully in control. It's very easy to create the mock, and all the methods area automatically implemented by PHPUnit.

Reply:
The idea here is to test that the definitions are sensible and we did not e.g. accidentally add an `object` type hint instead of `mixed` :) So basically this test tries to use it as it was intended. And this is also kind of a documentation-like example briefly showing with code how this interface can be used. Also by creating stub we prove that we can actually implement this interface in all PHP versions we support.

This seems to add much more confidence than just testing instantiation.

 Mocking here seems less ideal because of the less strict type checking mentioned above (+ who knows  what other differences from real classes), and it would be less convenient to define all methods via mock functions (specifying names via strings, etc., instead of just generating class via IDE).

Reply:
> The idea here is to test that the definitions are sensible and we did not e.g. accidentally add an object type hint instead of mixed :)

### TL;DR
This verification is already done by the signatures, and additional testing is redundant.

### Detail
Ah, I see now. Well, my idea for these tests was different: just to check whether signatures are compatible, and to provide a way to verify this should the interfaces change. I do not believe that testing whether a method that returns `object` declares that return type, because this is already a hard requirement of the interface, and the test would simply re-declare that. I do understand a different point of view, however: that design is reflected in a spec, and tests are an implementation of that spec, and therefore they should test whether the code complies with that spec. But this is overkill. IMO this would be acceptable if such tests were created separately and _automatically_. But in that case, you could just generate the interfaces from that same spec, and they would do that validation, so back to square one. Therefore, I would prefer to not do this. I used to do this before, but it really is time-consuming for no good reason IMO. Feel free to create a discussion, though.

Reply:
Yeah, since phpcs/psalm already parse and check the code for all PHP versions, these tests are probably not really needed.

The instantiation tests however do not seem to do anything useful too here, but added them for now.

Reply:
Thanks!